src/lib/common/string.cpp

```diff
@@ -676,3 +676,45 @@ void strReplace(char* to, int toLen, const char* from, const char* oldString, co
 
   to[toIx] = 0;
 }
+
+
+
+/* ****************************************************************************
+*
+* servicePathCheck - check one single component of the service-path
+*
+* NOTE
+* This function is used for register/notify only. '#' is not an accepted character,
```

Comment:
We shouldn't let the cases pass. Don't we have any functional test to ensure that?